src/ray/gcs/gcs_server/gcs_worker_manager.cc

```diff
@@ -25,87 +25,106 @@ void GcsWorkerManager::HandleReportWorkerFailure(
     rpc::SendReplyCallback send_reply_callback) {
   const rpc::Address worker_address = request.worker_failure().worker_address();
   const auto worker_id = WorkerID::FromBinary(worker_address.worker_id());
-  const auto node_id = NodeID::FromBinary(worker_address.raylet_id());
-  std::string message =
-      absl::StrCat("Reporting worker exit, worker id = ",
-                   worker_id.Hex(),
-                   ", node id = ",
-                   node_id.Hex(),
-                   ", address = ",
-                   worker_address.ip_address(),
-                   ", exit_type = ",
-                   rpc::WorkerExitType_Name(request.worker_failure().exit_type()),
-                   ", exit_detail = ",
-                   request.worker_failure().exit_detail());
-  if (request.worker_failure().exit_type() == rpc::WorkerExitType::INTENDED_USER_EXIT ||
-      request.worker_failure().exit_type() == rpc::WorkerExitType::INTENDED_SYSTEM_EXIT) {
-    RAY_LOG(DEBUG) << message;
-  } else {
-    RAY_LOG(WARNING) << message
-                     << ". Unintentional worker failures have been reported. If there "
-                        "are lots of this logs, that might indicate there are "
-                        "unexpected failures in the cluster.";
-  }
-  auto worker_failure_data = std::make_shared<WorkerTableData>();
-  worker_failure_data->CopyFrom(request.worker_failure());
-  worker_failure_data->set_is_alive(false);
-
-  for (auto &listener : worker_dead_listeners_) {
-    listener(worker_failure_data);
-  }
+  GetWorkerInfo(
```

Comment:
Are these changes mainly on the refactoring of the GetWorkerInfo funcion?

Reply:
yes. But also there's a new change that we get WorkerData before we update the failureData. Previously, we overwrote the whole data. Now we merge them.